bin/header_scraper.py

```diff
@@ -0,0 +1,150 @@
+        header_list = client.get(HEADERS_URL)
+        header_soup = BeautifulSoup(header_list.content, 'lxml')
+        rows = header_soup.find_all('table', class_='t-dsc-begin')[0].find_all('tr')
+
+        # assemble list of URLs from provided href tags
+        url_list = []
+        for row in rows:
+            link = row.find('a').get('href')
+            url_list.append(BASE_URL + link)
+
+        for url in url_list:
+
+            # scrape target URL
+            result = client.get(url)
+
+            # get HTML content as soup object
+            soup = BeautifulSoup(result.content, 'lxml')
+
+            # get all tables from relevant part of page
+            # only search direct children so as not to include Navbar
+            # exclude first element, which is TOC
+            tables = soup.find('div', id='mw-content-text').find_all('table', recursive=False)[1:]
+
+            for table in tables:
+                rows = table.find_all('tr')
+                for row in rows:
+                    data = row.find_all('td')
+
+                    # skip empty data
+                    if len(data) == 0:
+                        continue
+
+                    # categorize into header
+                    if len(data) < 2:
+                        # exclude non-library headings
+                        if 'header' in data[0].get_text():
+                            name = re.search('\w+\.h', data[0].get_text()).group(0)
+                            if name not in libraries.keys():
+                                header = Header()
+                                libraries[name] = header
+                            else:
+                                header = libraries[name]
+                        continue
+
+                    # data[0] will be the function, macro, etc.
+                    # data[1] will be the description and type (e.g. 'function', 'macro constant', etc.)
+                    type = data[1].get_text()
+                    # get span element containing all entries in this data block
+                    span = data[0].find('span')
+                    if not span:
+                        continue
+                    # iterate over individual elements' spans
+                    text = ''
+                    for element in span.find_all('span'):
+                        text += element.get_text().strip() + '\n'
+                    text = text.splitlines()
+
+                    # don't insert duplicates
+                    if text in seen:
+                        continue
+                    if '(function)' in type or '(function macro)' in type:
+                        header.appendf(text)
+                    elif '(macro constant)' in type:
+                        header.appendm(text)
+                    elif '(typedef)' in type:
+                        header.appendt(text)
+
+        # send output to YAML file
+        yaml_format(libraries)
+
+def yaml_format(libraries):
```

Comment:
Best to use a YAML library so that you don't need to worry about escaping anything.